app/src/main/java/org/simple/clinic/widgets/qrcodescanner/BitmapUtils.kt

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright 2020 Google LLC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.simple.clinic.widgets.qrcodescanner
+
+import android.annotation.TargetApi
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.graphics.ImageFormat
+import android.graphics.Matrix
+import android.graphics.Rect
+import android.graphics.YuvImage
+import android.media.Image.Plane
+import android.os.Build.VERSION_CODES
+import androidx.annotation.RequiresApi
+import androidx.camera.core.ExperimentalGetImage
+import androidx.camera.core.ImageProxy
+import timber.log.Timber
+import java.io.ByteArrayOutputStream
+import java.nio.ByteBuffer
+
+/** Utils functions for bitmap conversions.  */
+object BitmapUtils {
+
+  /** Converts NV21 format byte buffer to bitmap.  */
+  private fun getBitmap(data: ByteBuffer, metadata: FrameMetadata): Bitmap? {
+    data.rewind()
+    val imageInBuffer = ByteArray(data.limit())
+    data[imageInBuffer, 0, imageInBuffer.size]
+    try {
+      val image = YuvImage(
+          imageInBuffer, ImageFormat.NV21, metadata.width, metadata.height, null)
+      val stream = ByteArrayOutputStream()
+      image.compressToJpeg(Rect(0, 0, metadata.width, metadata.height), 80, stream)
+      val bmp = BitmapFactory.decodeByteArray(stream.toByteArray(), 0, stream.size())
+      stream.close()
+      return rotateBitmap(bmp, metadata.rotation, flipX = false, flipY = false)
+    } catch (e: Exception) {
+      Timber.e("Error: %s", e.message)
```

Comment:
We should be reporting this as a non-fatal error instead of pushing it to `Timber`.